README.md

```diff
@@ -1,2 +1,307 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: dhcp-server-conf
+data:
+  dhcpd.conf: |
+    authoritative;
+
+    default-lease-time 86400;
+    max-lease-time 86400;
+
+    subnet 192.168.123.0 netmask 255.255.255.0 {
+            range 192.168.123.2 192.168.123.254;
+            option broadcast-address 192.168.123.255;
+    }
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dhcp-server
+  labels:
+    app: dhcp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: dhcp
+  template:
+    metadata:
+      labels:
+        app: dhcp
+      annotations:
+        k8s.v1.cni.cncf.io/networks: '[
+          {
+            "name": "flatl2",
+            "ips": ["192.168.123.1/24"]
+          }
+        ]'
+    spec:
+      containers:
+      - name: server
+        image: quay.io/mduarted/dhcpd
+        args: ["-4", "-f", "-d", "--no-pid", "-cf", "/etc/dhcp/dhcpd.conf"]
+        securityContext:
+          runAsUser: 1000
+          privileged: true
+        volumeMounts:
+        - name: multus-daemon-config
+          mountPath: /etc/dhcp
+          readOnly: true
+      volumes:
+      - name: multus-daemon-config
+        configMap:
+          name: dhcp-server-conf
+          items:
+          - key: dhcpd.conf
+            path: dhcpd.conf
+EOF
+```
+
+The yaml above provisions a deployment (with a single replica) that will ensure
+the DHCP server is always up. Keep in mind this is a simple example and should
+**not** be used in production - there is a single point of failure (the DHCP
+server), and the leases file is not persisted outside the pod.
+
+The DHCP server configuration is quite simple, and is provided via a `ConfigMap`
+and mounted into the pod.
+
+It is also important to refer the network-admin **must** define a static IP
+address for the pod in the DHCP deployment, and ensure that IP is not advertised
+by the server.
+
+## Deploying the VM workloads
+Finally, the user should deploy the VM workloads. For that, execute the
+following command:
+```yaml
+cat <<EOF | kubectl apply -f -
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm-server
+spec:
+  running: true
+  template:
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - name: default
+            masquerade: {}
+          - name: flatl2-overlay
+            bridge: {}
+        machine:
+          type: ""
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+      - name: default
+        pod: {}
+      - name: flatl2-overlay
+        multus:
+          networkName: flatl2
+      terminationGracePeriodSeconds: 0
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: quay.io/kubevirt/fedora-with-test-tooling-container-disk:v1.0.0
```

Comment:
If there is a reason to prefer fedora over rhel here, why not using quay.io/containerdisks/fedora:38 ?

Reply:
pretty much any container img with ping would do. 

I guess I could improve the scenario with a container disk feature nginx or something like that ...

Do we have one w/ a webserver baked in ?

Reply:
No, it would have to be done via cloud-init, not nice.